### PR TITLE
Api 2.0

### PR DESCRIPTION
**Pull request by DNS320**
- add API changes, modify "Edit Connection" form to edit new settings
- add dark mode
- improve icons
- improve buttons

### CODE REVIEW

File: src/components/EditConnectionDialog.tsx [hunk 1]
```diff
@@ -69,86 +89,190 @@ const EditConnectionDialog: React.FC<Props> = ({
...
+                  true
+                }
+                helperText={formik.touched.delayIn && formik.errors.delayIn}
+              />
+              <TextField
+                margin="dense"
+                id="packetLossIn"
+                name="packetLossIn"
+                label="Packet Loss in %"
+                type="t"
```
Comment (DNS320) [inline src/components/EditConnectionDialog.tsx]: type="text"

File: src/components/EditConnectionDialog.tsx [hunk 2]
```diff
@@ -69,86 +89,190 @@ const EditConnectionDialog: React.FC<Props> = ({
...
+                helperText={
+                  formik.touched.packetLossIn && formik.errors.packetLossIn
+                }
+              />
+              <TextField
+                margin="dense"
+                id="jitterIn"
+                name="jitterIn"
+                label="Jitter in ms"
+                type="t"
```
Comment (DNS320) [inline src/components/EditConnectionDialog.tsx]: type="text"

File: src/components/EditConnectionDialog.tsx [hunk 3]
```diff
@@ -69,86 +89,190 @@ const EditConnectionDialog: React.FC<Props> = ({
...
+                  true
+                }
+                helperText={formik.touched.jitterIn && formik.errors.jitterIn}
+              />
+              <TextField
+                margin="dense"
+                id="bandwidthIn"
+                name="bandwidthIn"
+                label="Bandwidth in Mbit/s"
+                type="t"
```
Comment (DNS320) [inline src/components/EditConnectionDialog.tsx]: type="text"

File: src/components/EditConnectionDialog.tsx [hunk 4]
```diff
@@ -69,86 +89,190 @@ const EditConnectionDialog: React.FC<Props> = ({
...
+                )} 🡆 ${getLogicalInterfaceNameById(
+                  connection.secondLogicalInterfaceId,
+                )}`}
+              </DialogContentText>
+              <TextField
+                margin="dense"
+                id="delayIn"
+                name="delayIn"
+                label="Delay in ms"
+                type="t"
```
Comment (DNS320) [inline src/components/EditConnectionDialog.tsx]: type="text"

File: src/components/EditConnectionDialog.tsx [hunk 5]
```diff
@@ -69,86 +89,190 @@ const EditConnectionDialog: React.FC<Props> = ({
...
+                )} 🡄 ${getLogicalInterfaceNameById(
+                  connection.secondLogicalInterfaceId,
+                )}`}
+              </DialogContentText>
+              <TextField
+                margin="dense"
+                id="delayOut"
+                name="delayOut"
+                label="Delay in ms"
+                type="t"
```
Comment (DNS320) [inline src/components/EditConnectionDialog.tsx]: type="text"

File: src/components/EditConnectionDialog.tsx [hunk 6]
```diff
@@ -69,86 +89,190 @@ const EditConnectionDialog: React.FC<Props> = ({
...
+                  true
+                }
+                helperText={formik.touched.delayOut && formik.errors.delayOut}
+              />
+              <TextField
+                margin="dense"
+                id="packetLossOut"
+                name="packetLossOut"
+                label="Packet Loss in %"
+                type="t"
```
Comment (DNS320) [inline src/components/EditConnectionDialog.tsx]: type="text"

File: src/components/EditConnectionDialog.tsx [hunk 7]
```diff
@@ -69,86 +89,190 @@ const EditConnectionDialog: React.FC<Props> = ({
...
+                helperText={
+                  formik.touched.packetLossOut && formik.errors.packetLossOut
+                }
+              />
+              <TextField
+                margin="dense"
+                id="jitterOut"
+                name="jitterOut"
+                label="Jitter in ms"
+                type="t"
```
Comment (DNS320) [inline src/components/EditConnectionDialog.tsx]: type="text"

File: src/components/EditConnectionDialog.tsx [hunk 8]
```diff
@@ -69,86 +89,190 @@ const EditConnectionDialog: React.FC<Props> = ({
...
+                  true
+                }
+                helperText={formik.touched.jitterOut && formik.errors.jitterOut}
+              />
+              <TextField
+                margin="dense"
+                id="bandwidthOut"
+                name="bandwidthOut"
+                label="Bandwidth in Mbit/s"
+                type="t"
```
Comment (DNS320) [inline src/components/EditConnectionDialog.tsx]: type="text"

**Comment by severindellsperger** [comment]
Looks good I will merge it into the main/master branch.

**Pull request merged**